### PR TITLE
TINY-8235: Convert plugins D-I to use the new options API

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-8235

Description of Changes:

Second PR in the set of 5 or 6 PRs to convert all the plugins/themes to use the new Options API. As part of this I did make some changes to the `forced_plugins` as it was missed in my original PR and it could have been either a `string` or `string[]` so this normalises it.

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by ashfordneil — APPROVED** [review]

**Review by metricjs — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/EditorOptions.ts [hunk 1]
```diff
@@ -187,7 +196,7 @@ const processDefaultValue = <T, U>(name: string, defaultValue: T, processor: Pro
 
 const create = (editor: Editor, initialOptions: Record<string, unknown>): Options => {
   const registry: Record<string, OptionSpec<any, any>> = {};
-  const values: Record<string, any> = {};
+  const values: Record<string, { value: any; isDefault: boolean }> = {};
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/api/EditorOptions.ts:199]: I'm guessing this needs rebasing against the other branch?

File: modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const option: {
+  <K extends keyof EditorOptions>(name: K): (editor: Editor) => EditorOptions[K] | undefined;
+  <T>(name: string): (editor: Editor) => T | undefined;
+} = (name: string) => (editor: Editor) =>
+  editor.options.get(name);
+
+const register = (editor: Editor): void => {
+  const registerOption = editor.options.register;
+
+  const stringListProcessor = (value: unknown) => {
```
Comment (spocke) [inline modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts:22]: This one is a candidate for a core processor it's fairly common that we have lists like this.

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/EditorOptions.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/EditorOptions.ts:199]: Yeah it does... I haven't done that yet sorry as I only merged it about an hour ago.

File: modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts:22]: Honestly there's a few I've noticed that could be converted to built-in processors, this is definitely one and so is `object[]`.

Comment (TheSpyder) [reply modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts:22]: Good to know we're all on the same page (even if that means we're making the same comment on multiple PRs).

File: modules/tinymce/src/core/test/ts/browser/options/NormalizeOptionsTest.ts [hunk 3]
```diff
@@ -284,31 +284,31 @@ describe('browser.tinymce.core.options.NormalizeOptionsTest', () => {
     it('Merged settings forced_plugins in default override forced_plugins in user settings', () => {
       assert.deepEqual(
         NormalizeOptions.combineOptions(false, false, {}, { forced_plugins: [ 'a' ] }, { forced_plugins: [ 'b' ] }),
-        { external_plugins: {}, forced_plugins: [ 'b' ], plugins: 'a' },
+        { external_plugins: {}, forced_plugins: [ 'a' ], plugins: 'a' },
```
Comment (TheSpyder) [inline modules/tinymce/src/core/test/ts/browser/options/NormalizeOptionsTest.ts:287]: Was this a bug you just fixed? 🙂

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/options/NormalizeOptionsTest.ts (hunk 3 above)
Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/options/NormalizeOptionsTest.ts:287]: Technically yes? The `editor.settings` had the wrong value for `forced_plugins` but it still worked as intended for the `plugins` setting.

File: modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts:22]: As discussed at stand up today we've made the `string[]` allow both a string array and a comma/space separated string.

File: modules/tinymce/src/core/main/ts/api/EditorOptions.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/EditorOptions.ts]: This has been rebased now.

**Pull request merged**